src/main/kotlin/inc/evil/d17map/security/jwt/JWTFilter.kt

```diff
@@ -0,0 +1,53 @@
+package inc.evil.d17map.security.jwt
+
+
+import inc.evil.d17map.security.user.BasicUserDetailsService
+import jakarta.servlet.FilterChain
+import jakarta.servlet.ServletException
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.springframework.context.ApplicationContext
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource
+import org.springframework.stereotype.Component
+import org.springframework.web.filter.OncePerRequestFilter
+import java.io.IOException
+
+@Component
+class JWTFilter(
+    private val jwtService: JWTService,
+    private val applicationContext: ApplicationContext
+) : OncePerRequestFilter() {
+
+    @Throws(ServletException::class, IOException::class)
+    override fun doFilterInternal(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        filterChain: FilterChain
+    ) {
+        val authHeader = request.getHeader("Authorization")
+        var token: String? = null
+        var username: String? = null
```

Comment:
Może warto zrobić token i username jako Optional? 

` val token: Optional<String> = Optional.ofNullable(authHeader)
            .filter { it.startsWith("Bearer ") }
            .map { it.substring(7) }`
            
  `val username: Optional<String> = token.map { jwtService.extractUsername(it) }`

Reply:
Potem zamiast token != null można to zapisać jako token.isPresent

Reply:
Myślę, że można pomyśleć o przepisaniu tego kodu na bardziej Kotlinowy, ale akurat Optionale zostały zastąpione przez null safety w Kotlinie. Ale podoba mi się pomysł, aby użyć funkcji `filter` i `map`!

Reply:
Czy teraz jest bardziej kotlinowy??